SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialEventTracer.h

```diff
@@ -0,0 +1,108 @@
+// .cpp file, when the LatencyTracer is enabled?)
+// TODO(EventTracer): make sure the overhead of SpatialEventTracer is minimal when it's switched off
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialEventTracer, Log, All);
+
+// TODO - Individual RPC Calls (distinguishing between GDK and USER)
+// TODO - RPCs on newly created objects go through a different flow and need to be handled
+
+namespace worker
+{
+namespace c
+{
+struct Io_Stream;
+struct Trace_Item;
+} // namespace c
+} // namespace worker
+
+// Note(EventTracer): EventTracer must be created prior to WorkerConnection, since it has to be passed to ConnectionConfig
+// (see SpatialConnectionManager diff)
+
+namespace SpatialGDK
+{
+// Note: SpatialEventTracer wraps Trace_EventTracer related functionality
+// It is owned by the connection handler and ViewCoordinator.
+// SpatialNetDriver initializes SpatialSender and SpatialReceiver with pointers to EventTracer read from SpatialWorkerConnection.
+// Note(EventTracer): SpatialEventTracer is supposed to never be null in SpatialWorkerConnection, SpatialSender, SpatialReceiver. Make sure
+// there are necessary nullptr checks if that changes.
+
+class SpatialEventTracer
+{
+public:
+	SpatialEventTracer(const FString& WorkerId);
+	~SpatialEventTracer();
+
+	const worker::c::Trace_EventTracer* GetConstWorkerEventTracer() const { return EventTracer; };
+	worker::c::Trace_EventTracer* GetWorkerEventTracer() const { return EventTracer; }
+
+	// TODO(EventTracer): add the option to SpatialScopedActiveSpanId for Sent TraceEvents.
+	// Consider making sure it's not accepting rvalue (since SpatialSpanIdActivatNetDriveror must live long enough for the worker sent op to
+	// be registered with this SpanId) e.g. void TraceEvent(... SpatialScopedActiveSpanId&& SpanIdActivator) = delete;
+	// TODO(EventTracer): Communicate to others, that SpatialScopedActiveSpanId must be creating prior to calling worker send functions
+
+	TOptional<Trace_SpanId> TraceEvent(FSpatialTraceEvent SpatialTraceEvent);
+	TOptional<Trace_SpanId> TraceEvent(FSpatialTraceEvent SpatialTraceEvent, const worker::c::Trace_SpanId Causes);
+	TOptional<Trace_SpanId> TraceEvent(FSpatialTraceEvent SpatialTraceEvent, const TArray<worker::c::Trace_SpanId>& Causes);
+	TOptional<Trace_SpanId> TraceEvent(FSpatialTraceEvent SpatialTraceEvent, const worker::c::Trace_SpanId* Causes, int32 NumCauses);
+
+	bool IsEnabled() const;
+
+	void ComponentAdd(const Worker_Op& Op);
+	void ComponentRemove(const Worker_Op& Op);
+	void ComponentUpdate(const Worker_Op& Op);
+
+	bool GetSpanId(const EntityComponentId& Id, const uint32 FieldId, worker::c::Trace_SpanId& CauseSpanId, bool bRemove = true);
+	bool GetMostRecentSpanId(const EntityComponentId& Id, worker::c::Trace_SpanId& CauseSpanId, bool bRemove = true);
+
+	bool DropSpanId(const EntityComponentId& Id, const uint32 FieldId);
+	bool DropSpanIds(const EntityComponentId& Id);
+
+private:
+	bool bEnabled{ false };
```

Comment:
Nit - I think we prefer = rather than {} for primitives. Although there is a constructor so these could also all be set there.